### ISSUE TITLE
FF browser crashed and now all dials are missing and GSD is starting from scratch. How do I recover my dials and groups?

### ISSUE THREAD

**Issue by rwfoster2021** [fastaddons/GroupSpeedDial#79]
FF browser crashed and now all dials are missing and GSD is starting from scratch. How do I recover my dials and groups? Win 10 FF 93.0 
**Describe the bug**

**Steps To Reproduce**

**Expected behavior**

**Device info:**
 - operating system: 
 - Browser and version: 
 - GroupSpeedDial version: 

**Additional info**
Add any other info about the problem here.

**Comment by fastaddons** [comment]
How did it happened?
Any chance that you used "Firefox Reset"? Firefox sometimes offers it to fix issues, but it also clears all data.
If you used Firefox Reset, then you should have your old profile folder backed up on your desktop.
Also there are backups in the Firefox Sync. storage, so if were logged in your Firefox account, you should see a backup here:

![history](https://user-images.githubusercontent.com/55486931/140274423-3802173b-c1a0-4da3-b57c-77b43751ec12.png)

**Comment by Jayy77** [comment]
Hello,
I have the same problem since 2 days with GSD 16.8 Addon in FF 94.0.1 under W10 pro Version 21H1.
Was this problem solved and if yes where are dials stored so I can recover the dials?
I can't find the menue of the above picture.
I would be very happy if I could restore my 4 Tabs full of Speed Dials of this great Addon!
Thanks
Jochen

**Comment by fastaddons** [comment]
The backups are accessible from the main menu on the dials page - in the upper-right corner:

![_history](https://user-images.githubusercontent.com/55486931/143021900-f9bdc122-06ae-461c-b07a-23a461fe2708.png)

Also, what exactly happened? Usually data won't just disappear, maybe you've click Firefox Reset button?

**Comment by Jayy77** [comment]
![grafik](https://user-images.githubusercontent.com/94905421/143034296-9a289e1d-4276-49fa-a6a8-4d18579d57bd.png)

Thanks for the fast reply, this menue is not accessible since GSD is starting from scratch with asking for design and matrix. I am afraid to overwrite existing settings if I complete this??
No reset since all other addons are working as usual but maybe a Firefox crash 4 days ago has contributed. But the missing dials are only 1-2 days old.

**Comment by fastaddons** [comment]
So the data were lost but my addon was still installed? Or did you have to install it again?

You can complete the setup, it won't overwrite the data. After it's completed, try the History menu.
There are also some other backup options, but let's try the History first.

**Comment by Jayy77** [comment]
without me doing anything all GSD data was lost and whenever I opened a new empty tab, GSD wanted to go through the initialising routine.
Now I completed this setup with a raised pulse ;-) but afterwards found the changelog and could go back to it and apart from lost thumbnails that can easily be regenerated everything is back to normal. Thank you!!!!
I just wanted to make use of your paypal donation but the link in FF "about:addons" is not working. Where can I find a working one? Have a nice evening!

**Comment by fastaddons** [comment]
Oh, I'm so glad to hear it worked!
I've spend a lot of time implementing 3 different backup mechanisms so every time it helps someone with the recovery I'm super happy :).
Regarding the donation, you can buy me a coffee here :)
https://ko-fi.com/fastaddons
Or use the Paypal directly: https://www.paypal.com/paypalme/fastaddons
Thank you!

**Comment by Jayy77** [comment]
You really wrote an immensely useful and well layed out addon and offer it for free with even extremely quick support. That is just great thanks so much and almost sad, that the only thing not working with your programming is the donation link ;-)

**Comment by fastaddons** [comment]
Wow, that was a big donation! Thank you for supporting my work!
If you would like to get a Pro version of my addon, just create an cloud account on my cloud server and I will upgrade you right now.

**Comment by Jayy77** [comment]
you are welcome and your work deserves it. Thanks for the offer with the pro version. Maybe that is another thing to improve in your software to make users aware that there is a commercial pro version which I did not know. I have to read throgh the features and if I need them I am happy to pay the normal fee for it. From first glance at https://fastaddons.com/ it looks like it is mostly for generating private online dials to be used on different devices so I could see the same dials on my PC and mobile?

**Comment by fastaddons** [comment]
Yeah, my marketing is terrible, not many people knows about the Pro version since most features is available for free :).
I will be working on it later on, after I finish upgrading my server which I've been working for a several months. And then there is a lot of new features planned, it's gonna be great! I'm very excited.

Regarding my cloud server - the main feature is automatic backup / synchronization, so your dials will be always backed up in the cloud and synchronized across your devices / browsers (like work PC and home PC, at least that's how I use it and that's why I've build it). Another big feature is the "Cloud dials" web-page which works on every browser even without extension installed - so it works also in the phone browser. And you can enable Thumbnails synchronization which will synchronize also thumbnails across devices and you will see them in the cloud dials page.

Anyway, the full list of Pro features is here:
https://github.com/fastaddons/GroupSpeedDial/wiki/PRO-version-features

**Comment by Jayy77** [comment]
great features! and really bad marketing if not everybody knows about this possibility! maybe you should start a youtube channel to make more people aware of it... the only thing about this service being hosted on your server would be some privacy concerns so maybe a version that users can store on their own websites or even better a plugin to use online available GSDs hosted on their own Synology NAS ! There are millions of them and many of them could use your feature (as I would like this on my Synology) and if you would charge just a few dollars you could make some wealth...

**Comment by fastaddons** [comment]
I know, but it's not that easy. Currently the biggest issue is time - I've build so many addons and I have to spend some some time on each one from time to time, so right now I don't know what to do first.
You are right about the server, I'm planning to add support for custom servers / Google drive and other providers. I will also add end-to-end (E2E) encryption.
Even though the data are stored on a private server, it's well protected, hosted at professional provider (Tilaa.com) and I'm the only person with access. I'm taking user data privacy very seriously and I'm doing my best to protect the data.

Anyway, it's time to sleep! Good night :)
